content/en/blog/_posts/2020-03-30-topology-manager-beta.md

```diff
@@ -0,0 +1,517 @@
+
+
+```
+{cpu: {0, 1}, gpu: 0, nic: 0}
+```
+
+![drawing](/images/blog/2020-03-25-kubernetes-1.18-release-announcement/numa-hint-provider1.png)
+
+When considering **<code>Container1</code>** these resources are then presumed to be unavailable, and thus only the following set of hints will be generated:
+
+	
+	cpu: {{01: True}, {10: True}, {11: False}}
+	gpu-vendor.com/gpu: {{10: True}}
+	nic-vendor.com/nic: {{10: True}}
+
+With a resulting aligned allocation of:
+
+
+```
+{cpu: {4, 5}, gpu: 1, nic: 1}
+```
+
+![drawing](../../../../static/images/blog/2020-03-25-kubernetes-1.18-release-announcement/numa-hint-provider2.png)
+
+**NOTE:**Unlike the pseudocode provided at the beginning of this section, the call to **<code>Allocate()</code>** does not actually take a parameter for the merged “best” hint directly. Instead, the <strong><code>TopologyManager</code></strong> implements the following <strong><code>Store</code></strong> interface that <strong><code>HintProviders</code></strong> can query to retrieve the hint generated for a particular container once it has been generated:
+
+
+```
+type Store interface {
+    GetAffinity(podUID string, containerName string) TopologyHint
+}
+```
+
+
+Separating this out into its own API call allows one to access this hint outside of the pod admission loop. This is useful for debugging as well as for reporting generated hints in tools such as **<code>kubectl</code>**(not yet available).
+
+### Policy.Merge
+
+The merge strategy defined by a given policy dictates how it combines the set of **<code>TopologyHints</code>** generated by all <strong><code>HintProviders</code></strong> into a single <strong><code>TopologyHint</code></strong> that can be used to inform aligned resource allocations.
+
+The general merge strategy for all supported policies begins the same:
+
+1. Take the cross-product of **<code>TopologyHints</code>** generated for each resource type
+2. For each entry in the cross-product, <strong><code>bitwise-and</code></strong> the NUMA affinities of each <strong><code>TopologyHint</code></strong> together. Set this as the NUMA affinity in a resulting “merged” hint.
+3. If all of the hints in an entry have <strong><code>Preferred</code></strong> set to <strong> <code>True</code></strong> , set <strong><code>Preferred</code></strong> to <strong><code>True</code></strong> in the resulting “merged” hint.
+4. If even one of the hints in an entry has <strong><code>Preferred</code></strong> set to <strong> <code>False</code></strong> , set <strong><code>Preferred</code></strong> to <strong> <code>False</code></strong>  in the resulting “merged” hint. Also set <strong><code>Preferred</code></strong> to <strong><code>False</code></strong> in the “merged” hint if its NUMA affinity contains all 0s.
+
+Following the example from the previous section with hints for <strong><code>Container0</code></strong> generated as:
+
+<code>
+  cpu: {{01: True}, {10: True}, {11: False}}
```

Comment:
In addition to @mrbobbytables comment, can you also please indent the text for  `{{01: True}, {10: True}, {11: False}}` to line up with the similar text below it